File: etl-common/src/test/scala/org/bigdate/etl/common/scala/test/spark/executors/sink/FileSinkExecutor.scala
package org.bigdate.etl.common.scala.test.spark.executors.sink

import java.util

import org.apache.spark.SparkContext
import org.apache.spark.rdd.RDD
import org.bigdata.etl.common.annotations.ETLExecutor
import org.bigdata.etl.common.configs.FileConfig
import org.bigdata.etl.common.executors.SinkExecutor
import org.bigdate.etl.common.scala.test.spark.executors.source.FileSourceExecutor
import org.slf4j.{Logger, LoggerFactory}

/**
 * Author: GL
 * Date: 2022-05-05
 */
@ETLExecutor("file")
class FileSinkExecutor extends SinkExecutor[SparkContext, RDD[Integer], FileConfig] {

  val log: Logger = LoggerFactory.getLogger(classOf[FileSourceExecutor])

  override def init(engine: SparkContext, config: FileConfig): Unit = {
    log.info("FileSinkExecutor init, config: {}", config)
  }

  override def process(data: util.Collection[RDD[Integer]], config: FileConfig): Unit = {
    log.info("FileSinkExecutor process, config: {}", config)
    val rdd: RDD[Integer] = data.stream.findFirst.get()
    rdd.saveAsTextFile(config.getPath)
  }

  override def close(engine: SparkContext, config: FileConfig): Unit = {
    log.info("FileSinkExecutorExecutor close, config: {}", config)
  }

  override def check(config: FileConfig): Boolean = {
    log.info("FileSinkExecutorExecutor check, config: {}", config)
    true
  }
}

File: etl-common/src/test/java/org/bigdata/etl/common/executors/sink/FileSinkExecutor.java
package org.bigdata.etl.common.executors.sink;

import lombok.extern.slf4j.Slf4j;
import org.apache.spark.SparkContext;
import org.apache.spark.rdd.RDD;
import org.bigdata.etl.common.annotations.ETLExecutor;
import org.bigdata.etl.common.configs.FileConfig;
import org.bigdata.etl.common.executors.SinkExecutor;

import java.util.Collection;
import java.util.Optional;


/**
 * Author: GL
 * Date: 2022-04-22
 */
@Slf4j
@ETLExecutor("file")
public class FileSinkExecutor implements SinkExecutor<SparkContext, RDD<String>, FileConfig> {

    @Override
    public void init(SparkContext engine, FileConfig config) {
        log.info("FileSinkExecutor init, config: {}", config);
    }

    @Override
    public void process(SparkContext engine, RDD<String> value, FileConfig config) {
        log.info("FileSinkExecutor process, config: {}", config);
        value.saveAsTextFile(config.getPath());
    }

    @Override
    public void close(SparkContext engine, FileConfig config) {
        log.info("FileSinkExecutorExecutor close, config: {}", config);
    }

    @Override
    public boolean check(FileConfig config) {
        log.info("FileSinkExecutorExecutor check, config: {}", config);
        return true;
    }
}
